pkg/kudoctl/resources/dependencies/resolve.go

```diff
@@ -79,15 +87,34 @@ func dependencyWalk(
 	g *dependencyGraph,
 	parent packages.Resources,
 	parentIndex int,
-	resolver pkgresolver.Resolver) error {
+	resolver pkgresolver.Resolver,
+	operatorDirectory *string) error {
 	//nolint:errcheck
 	childrenTasks := funk.Filter(parent.OperatorVersion.Spec.Tasks, func(task kudoapi.Task) bool {
 		return task.Kind == engtask.KudoOperatorTaskKind
 	}).([]kudoapi.Task)
 
 	for _, childTask := range childrenTasks {
+		childPackageName := childTask.Spec.KudoOperatorTaskSpec.Package
+
+		// if the path to a child dependency is a relative one, we construct the absolute path for the
+		// resolver by combining the absolute path for the operator directory with the relative dependency path
+		// a relative dependency path must begin with either './' or '../'
+		isRelativePackage := strings.HasPrefix(childPackageName, "./") || strings.HasPrefix(childPackageName, "../")
```

Comment:
We could also check `os.Stat(filepath.Join(*operatorDirectory, childPackageName))`. With that we won't require that paths start with `./` or `../`. Though we need to check that `childPackageName` isn't an absolute path before. A function `isRelativePackage` could do that. WDYT?

Reply:
> Though we need to check that childPackageName isn't an absolute path before.

My initial implementation was similar but it is just more cumbersome. Furthermore, that's what we do for local packages [anyway](https://github.com/kudobuilder/kudo/commit/f58547c58b9e6e1a9241ffc4e7e5f4326efe1090) - e.g. `kudo install zookeeper` will **not** take the local _zookeeper_ folder - for that you need to reference it like `./zookeeper`

Reply:
I like the required prefixes with `./` and `../` - I think it's bad style to use a local path without any prefix. So 👍

Reply:
Yeah, no strong opinions here. Let's just make sure that this requirement is documented.

Reply:
I've [opened a PR](https://github.com/kudobuilder/kudo.dev/pull/298) to improve the docs.